noir-projects/noir-protocol-circuits/crates/types/src/type_serialization.nr

```diff
@@ -172,32 +172,90 @@ where
     }
 }
 
-#[test]
-fn test_u16_serialization() {
-    let a: u16 = 10;
-    assert_eq(a, u16::deserialize(a.serialize()));
-}
+impl<T, let N: u32> Serialize<N + 1> for Option<T>
+where
+    T: Serialize<N>,
+{
+    fn serialize(self) -> [Field; N + 1] {
+        let mut result: [Field; N + 1] = std::mem::zeroed();
```

Comment:
Just out of curiosity is the `std::mem::zeroed()` approach better in someway then just doing the following?

```
let mut result = [0; N + 1];
```

Reply:
They should be equivalent. I like `zeroed` becuse I find it makes the initialization more explicit, because it can be used for other types as well, and because it makes the array type (`Field`) more explicit as well. I don't like how `[0; N + 1]` combines values (`0`) with type (`[N + 1]`).